### ISSUE TITLE
[new operator] CompactSpaces

### ISSUE THREAD

**Issue by t9md** [t9md/atom-vim-mode-plus#279]
Compact multiple white spaces into single space
Convenient change aligned code format to non-aligned one.
- Don't touch leading and trailing spaces.

So convert like following. `_` represent single space in this example.
From `___a___b____c_` To `____a_b_c_`.(NOTE leading/trailing space is not changed).

**Comment by t9md** [comment]
![compact-spaces](https://cloud.githubusercontent.com/assets/155205/15363632/55ae15ce-1d54-11e6-9afb-89f35d33fe89.gif)

**Comment by t9md** [comment]
- [ ] Write spec

**Comment by t9md** [comment]
I once implemented to "skip commented row when target is linewise".
It's good when CompactSpaces to whole buffer. But not the other case.
For example, when user want to do space compaction on commented line by selecting lines of consecutive comment, this special feature prevent to do that. So I removed this once I thought good feature.

## NOTE to future me

This kind of "seem to good feature" is traps I should avoid.
Basically I should skip the feature which introduce ambiguity of operation, ambiguity, no-intuitivity prevent user to self learn by their operation. 

So user have to consult other resource and popping up same question again & again.
Is that really necessary discussion? Programmer have lot of things to learn and memory space is limited.
Don't consume their brain CPU and memory by knee-jerk-seem-to-good-feature.
